### PR TITLE
Add OTEL_EXPORTER_JAEGER_PROTOCOL and improve description of other OTEL_EXPORTER_JAEGER_* env vars

### PR DESCRIPTION
**Pull request by pellared**
Follow-up of #2333

## Changes

Preview [here](https://github.com/pellared/opentelemetry-specification/blob/patch-2/specification/sdk-environment-variables.md#jaeger-exporter)

- Add the `OTEL_EXPORTER_JAEGER_PROTOCOL` environment variable to select the protocol used
  by the Jaeger exporter.
- More explicit description of the rest of the  `OTEL_EXPORTER_JAEGER_*` environment variables (e.g. when they are applicable - it also represents the current state of SDKs)

Here are the following reasons for this change

1. Enable the possibility of selecting the protocol when the Jaeger exporter is set via env var `OTEL_TRACES_EXPORTER=jaeger` (instead of supporting values like `OTEL_TRACES_EXPORTER=jaeger-udp`). 
2. This resembles [the configuration of the OTLP exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) which has `OTEL_EXPORTER_OTLP_PROTOCOL`
3. The newest OTel .NET SDK RC version supports this env var. See [here](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Jaeger). I feel that it is safer to standardize it (especially the supported values) before it is released
4. We want to take advantage of it in [OTel .NET AutoInstrumentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation). However, I believe this could be useful in other places as well.
5. Node.js SDK uses `6832` as the default Agent port because it uses a different protocol

Related issues:
- https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/355

FYI (all SIG maintainers where I see that `OTEL_EXPORTER_JAEGER_ENDPOINT` is used) @open-telemetry/dotnet-maintainers @open-telemetry/dotnet-instrumentation-maintainers @open-telemetry/java-instrumentation-maintainers @open-telemetry/python-maintainers @open-telemetry/go-maintainers @open-telemetry/php-maintainers  @open-telemetry/java-maintainers @open-telemetry/ruby-maintainers @open-telemetry/rust-maintainers 

## Alternative

Instead of adding `OTEL_EXPORTER_JAEGER_PROTOCOL` we can specify more specific values for `OTEL_TRACES_EXPORTER` like `jaeger/http/thrift.binary` but this would be inconsistent with the OTLP exporter configuration.

## TODO in next PR

- reformatting the table to make it more readable
  - https://github.com/open-telemetry/opentelemetry-specification/pull/2341#discussion_r806627129
  - https://github.com/open-telemetry/opentelemetry-specification/pull/2341#discussion_r806965306
  - https://github.com/open-telemetry/opentelemetry-specification/pull/2341#discussion_r808263460

### CODE REVIEW

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -127,15 +127,21 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
+| OTEL_EXPORTER_JAEGER_ENDPOINT   | Full URL of endpoint for Jaeger traces [2] | `http://localhost:14268/api/traces` for `http/thrift.binary` protocol, `http://localhost:14250` for `grpc` protocol    |
 | OTEL_EXPORTER_JAEGER_TIMEOUT    | Maximum time the Jaeger exporter will wait for each batch export | 10s                                                                                              |
 | OTEL_EXPORTER_JAEGER_USER       | Username to be used for HTTP basic authentication                |                                                                                                  |
 | OTEL_EXPORTER_JAEGER_PASSWORD   | Password to be used for HTTP basic authentication                |                                                                                                  |
+| OTEL_EXPORTER_JAEGER_PROTOCOL   | The transport protocol. Options MAY include `http/thrift.binary`, `grpc`, `udp/thrift.compact` | `http/thrift.binary` [3]                                            |
```
[the remark this answers is not in the document]
Comment (pellared) [reply specification/sdk-environment-variables.md:135]: @yurishkuro I do not get the suggestion.

1. In https://github.com/open-telemetry/opentelemetry-specification/pull/2333 it is suggested to default to Thrift over HTTP (which is what `http/thift.binary` stands for
2. the `udp/thrift.binary` is not even listed as a possible option

Or maybe you want to simply mention `udp/thrift.binary` as a supported option?

```suggestion
| OTEL_EXPORTER_JAEGER_PROTOCOL   | The transport protocol. Options MAY include `http/thrift.binary`, `grpc`, `udp/thrift.compact`, `udp/thrift.binary` | `http/thrift.binary` [3]                                            |
```

Still, I will reorder and put it in at the top 👍

File: specification/sdk-environment-variables.md [hunk 2]
```diff
@@ -127,15 +127,21 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
 
-[1] See [Jaeger Agent](https://www.jaegertracing.io/docs/latest/deployment/#agent) documentation.
+[1] Used when `udp/thrift.compact` protocol is selected.
```
[the remark this answers is not in the document]
Comment (pellared) [reply specification/sdk-environment-variables.md:137]: @yurishkuro  .NET SDK supports UDP as well. I will check if Node.js SDK supports ``udp/thrift.binary` protocol. If so I will adopt this PR accordingly.

[the remark this answers is not in the document]
Comment (pellared) [reply specification/sdk-environment-variables.md:137]: You are correct. The Node.js supports `udp/thrift.binary`. I will make changes in this PR tomorrow. Thanks 👍

File: specification/sdk-environment-variables.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (yurishkuro) [reply specification/sdk-environment-variables.md:135]: sorry, ignore the edit suggestion, I misread the syntax that it's the next column in the table. But yes, `udp/thrift.binary` is a possible value (which implies a different port #)

File: specification/sdk-environment-variables.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pellared) [reply specification/sdk-environment-variables.md]: Addressed https://github.com/open-telemetry/opentelemetry-specification/pull/2341/commits/1b746ac4fa497619abce08cecec2b677383abfa9

File: specification/sdk-environment-variables.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pellared) [reply specification/sdk-environment-variables.md]: Addressed https://github.com/open-telemetry/opentelemetry-specification/pull/2341/commits/1b746ac4fa497619abce08cecec2b677383abfa9

File: specification/sdk-environment-variables.md [hunk 3]
```diff
@@ -126,16 +126,25 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
+| OTEL_EXPORTER_JAEGER_USER (`http/thrift.binary`, `grpc` protocols) | Username to be used for HTTP basic authentication |                                                                              |
+| OTEL_EXPORTER_JAEGER_PASSWORD (`http/thrift.binary`, `grpc` protocols) | Password to be used for HTTP basic authentication |                                                                          |
+| OTEL_EXPORTER_JAEGER_AGENT_HOST (`udp/thrift.*` protocols) | Hostname of the [Jaeger agent][jaeger_agent] | `localhost`                                                                               |
+| OTEL_EXPORTER_JAEGER_AGENT_PORT (`udp/thrift.compact` protocol) | `udp/thrift.compact` port of the [Jaeger agent][jaeger_agent] | `6831`                                                              |
+| OTEL_EXPORTER_JAEGER_AGENT_PORT (`udp/thrift.binary` protocol) | `udp/thrift.binary` port of the [Jaeger agent][jaeger_agent] | `6832`                                                                |
```
Comment (pellared) [inline specification/sdk-environment-variables.md:137]: I was considering to have sections for each protocol. I would rather do it in a separate PR as I found such change subjective - readability vs maintainability.

**Comment by pellared** [comment]
Ready for another review round.

**Review by yurishkuro — APPROVED** [review]

File: specification/sdk-environment-variables.md [hunk 4]
```diff
@@ -126,16 +126,25 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
-| OTEL_EXPORTER_JAEGER_USER       | Username to be used for HTTP basic authentication                |                                                                                                  |
-| OTEL_EXPORTER_JAEGER_PASSWORD   | Password to be used for HTTP basic authentication                |                                                                                                  |
+| OTEL_EXPORTER_JAEGER_PROTOCOL   | The transport protocol. Options MAY include [`http/thrift.binary`][jaeger_http], [`grpc`][jaeger_grpc], [`udp/thrift.compact`][jaeger_udp], [`udp/thrift.binary`][jaeger_udp] | `http/thrift.binary` [1] |
+| OTEL_EXPORTER_JAEGER_ENDPOINT (`http/thrift.binary` protocol) | Full URL of the [Jaeger HTTP endpoint][jaeger_collector] | `http://localhost:14268/api/traces`                                        |
+| OTEL_EXPORTER_JAEGER_ENDPOINT (`grpc` protocol) | URL of the [Jaeger gRPC endpoint][jaeger_collector] | `http://localhost:14250`                                                                      |
```
Comment (tigrannajaryan) [inline specification/sdk-environment-variables.md:131]: Listing the same env var twice is a bit confusing. Is there any way to fit this on one line or too much info? Maybe use nested-tables (if there is such thing in markdown)?

**Review by tigrannajaryan — COMMENTED** [review]
Side note: the formatting of this page makes Github diffs difficult to read. Perhaps we need to reformat it similar to https://github.com/open-telemetry/opentelemetry-specification/pull/2345 (can be done in a separate, future PR).

File: specification/sdk-environment-variables.md [hunk 5]
```diff
@@ -126,16 +126,25 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
-| OTEL_EXPORTER_JAEGER_ENDPOINT   | Full URL of [Thrift over HTTP][jaeger_http] endpoint for Jaeger traces [2] | <!-- markdown-link-check-disable --> `"http://localhost:14268/api/traces"` <!-- markdown-link-check-enable --> |
-| OTEL_EXPORTER_JAEGER_TIMEOUT    | Maximum time the Jaeger exporter will wait for each batch export | 10s                                                                                              |
-| OTEL_EXPORTER_JAEGER_USER       | Username to be used for HTTP basic authentication                |                                                                                                  |
-| OTEL_EXPORTER_JAEGER_PASSWORD   | Password to be used for HTTP basic authentication                |                                                                                                  |
+| OTEL_EXPORTER_JAEGER_PROTOCOL   | The transport protocol. Options MAY include [`http/thrift.binary`][jaeger_http], [`grpc`][jaeger_grpc], [`udp/thrift.compact`][jaeger_udp], [`udp/thrift.binary`][jaeger_udp] | `http/thrift.binary` [1] |
```
Comment (tigrannajaryan) [inline specification/sdk-environment-variables.md:129]: Why does this say `MAY`? This is not a recommendation, it is a strict list. Should it say: `MUST be one of ...` instead?.

File: specification/sdk-environment-variables.md (hunk 4 above)
Comment (pellared) [reply specification/sdk-environment-variables.md:131]: As notes on your summary and https://github.com/open-telemetry/opentelemetry-specification/pull/2341#discussion_r806627129 I would rather address it in a separate PR

**Review by tigrannajaryan — APPROVED** [review]

File: specification/sdk-environment-variables.md (hunk 4 above)
Comment (pellared) [reply specification/sdk-environment-variables.md:131]: I will create issues for it once this PR is merged and assign it to myself.

File: specification/sdk-environment-variables.md (hunk 3 above)
Comment (owais) [reply specification/sdk-environment-variables.md:137]: image

nit: This is quite hard to read with env vars duplicated and some not in the left column. Can we may be not duplicate and add proto specific info in cells on the right?

Comment (pellared) [reply specification/sdk-environment-variables.md:137]: There is another proposal to make it similar https://github.com/open-telemetry/opentelemetry-specification/pull/2345 . Let me address the formatting in a separate PR (it is already in the TODO list and I will create issue once this PR is merged).

File: specification/sdk-environment-variables.md (hunk 5 above)
Comment (MrAlias) [reply specification/sdk-environment-variables.md:129]: Addressing this addition in a future PR could potentially lead to compatibility issues. Reviewers of the future PR may not now if this change was released and "upgrading" from an option to a requirement would be a backwards incompatible change. Or, even worse, this would be released prior to the follow on PR being merged. Given this section of the doc is marked stable, this seems like a valid concern.

Can we address the change from `MAY` to `MUST` for these changes in this PR instead? That way it will be a cohesive and complete change that is not copying prior errors.

**Review by MrAlias — APPROVED** [review]

File: specification/sdk-environment-variables.md [hunk 6]
```diff
@@ -126,16 +126,25 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
...
-| OTEL_EXPORTER_JAEGER_USER       | Username to be used for HTTP basic authentication                |                                                                                                  |
-| OTEL_EXPORTER_JAEGER_PASSWORD   | Password to be used for HTTP basic authentication                |                                                                                                  |
+| OTEL_EXPORTER_JAEGER_PROTOCOL   | The transport protocol. The value MUST be one of: [`http/thrift.binary`][jaeger_http], [`grpc`][jaeger_grpc], [`udp/thrift.compact`][jaeger_udp], [`udp/thrift.binary`][jaeger_udp] | `http/thrift.binary` [1] |
+| OTEL_EXPORTER_JAEGER_ENDPOINT (`http/thrift.binary` protocol) | Full URL of the [Jaeger HTTP endpoint][jaeger_collector] | `http://localhost:14268/api/traces`                                        |
+| OTEL_EXPORTER_JAEGER_ENDPOINT (`grpc` protocol) | URL of the [Jaeger gRPC endpoint][jaeger_collector] | `http://localhost:14250`                                                                      |
```
Comment (yurishkuro) [inline specification/sdk-environment-variables.md:131]:
```suggestion
| OTEL_EXPORTER_JAEGER_ENDPOINT  | Full URL of the [Jaeger HTTP endpoint][jaeger_collector] when using `http/thrift.binary` protocol | `http://localhost:14268/api/traces`                                        |
|   | URL of the [Jaeger gRPC endpoint][jaeger_collector] when using `grpc` protocol | `http://localhost:14250`                                                                      |
```
```suggestion
| OTEL_EXPORTER_JAEGER_ENDPOINT (`http/thrift.binary` protocol) | Full URL of the [Jaeger HTTP endpoint][jaeger_collector] | `http://localhost:14268/api/traces`                                        |
| OTEL_EXPORTER_JAEGER_ENDPOINT (`grpc` protocol) | URL of the [Jaeger gRPC endpoint][jaeger_collector] | `http://localhost:14250`                                                                      |
```

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Ping @owais ;)

**Review by owais — APPROVED** [review]
